azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -385,10 +387,11 @@ private boolean canWriteInCurrentWorkingDirectory(final String effectiveUser)
    */
   private void assignUserDirOwnership(final String effectiveUser) throws IOException {
     final ExecuteAsUser executeAsUser = new ExecuteAsUser(
-        this.sysProps.getString(NATIVE_LIB_FOLDER));
+        this.sysProps.getString(AZKABAN_SERVER_NATIVE_LIB_FOLDER));
+    final String groupName = this.sysProps.getString(AZKABAN_SERVER_GROUP_NAME, "azkaban");
     final List<String> changeOwnershipCommand = Arrays
-        .asList(CHOWN, effectiveUser + ":" + GROUP_NAME, getWorkingDirectory());
-    info("Change current working directory ownership to " + effectiveUser + ":" + GROUP_NAME + ".");
+        .asList(CHOWN, effectiveUser + ":" + groupName, getWorkingDirectory());
```

Comment:
Possible direction: 
Have an OS Helper/Util class which leverages `ExecuteAsUser` class to execute OS level functions since we do plenty of that in Azkaban. We have an `OsMemoryUtil` for example.
```
// So OsHelper becomes our central place for doing these kinda stuff.
new OsHelper().chown(getWorkingDirectory(), user, group);

```
However, I don't think this needs to be done now.